From 32ee7f91945f76f223a85fd5ff81e177c3ad8686 Mon Sep 17 00:00:00 2001
From: Fabian Vogt <fvogt@suse.com>
Date: Mon, 23 Jul 2018 13:59:36 +0200
Subject: [PATCH] Add new palette with standard VGA colors

The standard VGA colors are used by fbcon by default as well, which means
this is now as close to the linux console as possible.
---
 src/tsm/tsm-vte.c |   24 ++++++++++++++++++++++++
 1 file changed, 24 insertions(+)

Index: libtsm-4.0.2/src/tsm/tsm-vte.c
===================================================================
--- libtsm-4.0.2.orig/src/tsm/tsm-vte.c
+++ libtsm-4.0.2/src/tsm/tsm-vte.c
@@ -343,6 +343,28 @@ static uint8_t color_palette_base16_ligh
 	[TSM_COLOR_BACKGROUND]    = { 0xd8, 0xd8, 0xd8 }, /* light grey */
 };
 
+static uint8_t color_palette_vga[TSM_COLOR_NUM][3] = {
+	[TSM_COLOR_BLACK]         = {   0,   0,   0 },
+	[TSM_COLOR_RED]           = { 170,   0,   0 },
+	[TSM_COLOR_GREEN]         = {   0, 170,   0 },
+	[TSM_COLOR_YELLOW]        = { 170,  85,   0 },
+	[TSM_COLOR_BLUE]          = {   0,   0, 170 },
+	[TSM_COLOR_MAGENTA]       = { 170,   0, 170 },
+	[TSM_COLOR_CYAN]          = {   0, 170, 170 },
+	[TSM_COLOR_LIGHT_GREY]    = { 170, 170, 170 },
+	[TSM_COLOR_DARK_GREY]     = {  85,  85,  85 },
+	[TSM_COLOR_LIGHT_RED]     = { 255,  85,  85 },
+	[TSM_COLOR_LIGHT_GREEN]   = {  85, 255,  85 },
+	[TSM_COLOR_LIGHT_YELLOW]  = { 255, 255,  85 },
+	[TSM_COLOR_LIGHT_BLUE]    = {  85,  85, 255 },
+	[TSM_COLOR_LIGHT_MAGENTA] = { 255,  85, 255 },
+	[TSM_COLOR_LIGHT_CYAN]    = {  85, 255, 255 },
+	[TSM_COLOR_WHITE]         = { 255, 255, 255 },
+
+	[TSM_COLOR_FOREGROUND]    = { 170, 170, 170 },
+	[TSM_COLOR_BACKGROUND]    = {   0,   0,   0 },
+};
+
 static uint8_t (*get_palette(struct tsm_vte *vte))[3]
 {
 	if (!vte->palette_name)
@@ -362,6 +384,8 @@ static uint8_t (*get_palette(struct tsm_
 		return color_palette_base16_dark;
 	if (!strcmp(vte->palette_name, "base16-light"))
 		return color_palette_base16_light;
+	if (!strcmp(vte->palette_name, "vga"))
+		return color_palette_vga;
 
 	return color_palette;
 }
